File: year2/semestrul2/MPP/Laboratoare/lab3_cscs/lab3_cscs/Model/Inscriere.cs
namespace lab3_cscs.Model
{
    public class Inscriere
    {
        public string IdInscriere { get; set; }
        public string IdParticipant { get; set; }
        public string IdProba { get; set; }
        public string DataInscriere { get; set; }

        public Inscriere(string idInscriere, string idParticipant, string idProba, string dataInscriere)
        {
            IdInscriere = idInscriere;
            IdParticipant = idParticipant;
            IdProba = idProba;
            DataInscriere = dataInscriere;
        }

        public override string ToString()
        {
            return $"Inscriere{{idInscriere='{IdInscriere}', idParticipant='{IdParticipant}', idProba='{IdProba}', dataInscriere='{DataInscriere}'}}";
        }

        public override bool Equals(object obj)
        {
            if (this == obj) return true;
            if (obj == null || GetType() != obj.GetType()) return false;
            Inscriere inscriere = (Inscriere)obj;
            return IdInscriere.Equals(inscriere.IdInscriere);
        }

        public override int GetHashCode()
        {
            return IdInscriere.GetHashCode();
        }
    }
}

File: year2/semestrul2/MPP/Laboratoare/lab3_java1/src/main/java/org/example/lab3_java1/Model/Inscriere.java
package org.example.lab3_java1.Model;

public class Inscriere {
    private String id_inscriere;
    private String id_participant;
    private String id_proba;
    private String data_inscriere;

    public Inscriere(String id_inscriere, String id_participant, String id_proba, String data_inscriere) {
        this.id_inscriere = id_inscriere;
        this.id_participant = id_participant;
        this.id_proba = id_proba;
        this.data_inscriere = data_inscriere;
    }

    public String getId_inscriere() {
        return id_inscriere;
    }

    public void setId_inscriere(String id_inscriere) {
        this.id_inscriere = id_inscriere;
    }

    public String getId_participant() {
        return id_participant;
    }

    public void setId_participant(String id_participant) {
        this.id_participant = id_participant;
    }

    public String getId_proba() {
        return id_proba;
    }

    public void setId_proba(String id_proba) {
        this.id_proba = id_proba;
    }

    public String getData_inscriere() {
        return data_inscriere;
    }

    public void setData_inscriere(String data_inscriere) {
        this.data_inscriere = data_inscriere;
    }

    @Override
    public String toString() {
        return "Inscriere{" +
                "id_inscriere='" + id_inscriere + '\'' +
                ", id_participant='" + id_participant + '\'' +
                ", id_proba='" + id_proba + '\'' +
                ", data_inscriere='" + data_inscriere + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Inscriere inscriere = (Inscriere) o;
        return id_inscriere.equals(inscriere.id_inscriere);
    }

    @Override
    public int hashCode() {
        return id_inscriere.hashCode();
    }


}
